easybuild/easyblocks/m/mxnet.py

```diff
@@ -0,0 +1,222 @@
+from easybuild.tools.filetools import rmtree2, symlink, write_file
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_MXNet(MakeCp):
+    """Easyblock to build and install MXNet"""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Change default values of options"""
+        extra = MakeCp.extra_options()
+        # files_to_copy is not mandatory here
+        extra['files_to_copy'][2] = CUSTOM
+        return extra
+
+    def __init__(self, *args, **kwargs):
+        """Initialize custom class variables."""
+        super(EB_MXNet, self).__init__(*args, **kwargs)
+
+        self.mxnet_src_dir = None
+        self.py_ext = PythonPackage(self, {'name': self.name, 'version': self.version})
+        self.py_ext.module_generator = self.module_generator
+        self.r_ext = RPackage(self, {'name': self.name, 'version': self.version})
+        self.r_ext.module_generator = self.module_generator
+
+    def extract_step(self):
+        """
+        Prepare a combined MXNet source tree. Move all submodules
+        to their right place.
+        """
+        # Extract everything into separate directories.
+        super(EB_MXNet, self).extract_step()
+
+        mxnet_dirs = glob.glob(os.path.join(self.builddir, 'mxnet-*'))
+        if len(mxnet_dirs) == 1:
+            self.mxnet_src_dir = mxnet_dirs[0]
+            self.log.debug("MXNet dir is: %s", self.mxnet_src_dir)
+        else:
+            raise EasyBuildError("Failed to find/isolate MXNet source directory: %s", mxnet_dirs)
+
+        for srcdir in os.listdir(self.builddir):
+            if not srcdir.startswith('mxnet-'):
+                submodule, _, _ = srcdir.rpartition('-')
+                newdir = os.path.join(self.mxnet_src_dir, submodule)
+                olddir = os.path.join(self.builddir, srcdir)
+                # first remove empty existing directory
+                rmtree2(newdir)
+                try:
+                    shutil.move(olddir, newdir)
+                except IOError, err:
+                    raise EasyBuildError("Failed to move %s to %s: %s", olddir, newdir, err)
+
+        # the nnvm submodules has dmlc-core as a submodule too. Let's put a symlink in place.
+        newdir = os.path.join(self.mxnet_src_dir, "nnvm", "dmlc-core")
+        olddir = os.path.join(self.mxnet_src_dir, "dmlc-core")
+        rmtree2(newdir)
+        symlink(olddir, newdir)
+
+    def configure_step(self):
+        """Patch 'config.mk' file to use EB stuff"""
+        copy_file('make/config.mk', '.')
+
+        regex_subs = [
+            (r"export CC = gcc", r"# \g<0>"),
+            (r"export CXX = g\+\+", r"# \g<0>"),
+            (r"(?P<var>ADD_CFLAGS\s*=)\s*$", r"\g<var> %s" % os.environ['CFLAGS']),
+            (r"(?P<var>ADD_LDFLAGS\s*=)\s*$", r"\g<var> %s" % os.environ['LDFLAGS']),
+        ]
```

Comment:
Howso? You're deriving from `MakeCp` and not customising the `build_step`, so this is standard `make`?